kyuubi-server/src/main/scala/org/apache/kyuubi/server/statestore/SessionStateStore.scala

```diff
@@ -0,0 +1,155 @@
+    SessionStateStore.set(null)
+    super.stop()
+  }
+
+  def insertMetadata(metadata: Metadata): Unit = {
+    _stateStore.insertMetadata(metadata)
+  }
+
+  def getBatch(batchId: String): Batch = {
+    Option(_stateStore.getMetadata(batchId, true)).map(buildBatch).orNull
+  }
+
+  def getBatchesByType(batchType: String, from: Int, size: Int): Seq[Batch] = {
+    _stateStore.getMetadataList(
+      SessionType.BATCH,
+      batchType,
+      null,
+      null,
+      null,
+      from,
+      size,
+      true).map(buildBatch)
+  }
+
+  def updateBatchMetadata(
+      batchId: String,
+      state: String,
+      batchAppStatus: Map[String, String],
+      endTime: Long): Unit = {
+    val appId = batchAppStatus.get(APP_ID_KEY).orNull
+    val appName = batchAppStatus.get(APP_NAME_KEY).orNull
+    val appUrl = batchAppStatus.get(APP_URL_KEY).orNull
+    val appState = batchAppStatus.get(APP_STATE_KEY).orNull
+    val appError = batchAppStatus.get(APP_ERROR_KEY)
+    val metadata = Metadata(
+      identifier = batchId,
+      state = state,
+      engineId = appId,
+      engineName = appName,
+      engineUrl = appUrl,
+      engineState = appState,
+      engineError = appError,
+      endTime = endTime)
+    _stateStore.updateMetadata(metadata)
+  }
+
+  def cleanupMetadataById(batchId: String): Unit = {
+    _stateStore.cleanupMetadataByIdentifier(batchId)
+  }
+
+  private def buildBatch(batchMetadata: Metadata): Batch = {
+    val batchAppInfo = Map(
+      APP_ID_KEY -> Option(batchMetadata.engineId),
+      APP_NAME_KEY -> Option(batchMetadata.engineName),
+      APP_STATE_KEY -> Option(batchMetadata.engineState),
+      APP_URL_KEY -> Option(batchMetadata.engineUrl),
+      APP_ERROR_KEY -> batchMetadata.engineError)
+      .filter(_._2.isDefined)
+      .map(info => (info._1, info._2.get))
+
+    new Batch(
+      batchMetadata.identifier,
+      batchMetadata.engineType,
+      batchAppInfo.asJava,
+      batchMetadata.kyuubiInstance,
+      batchMetadata.state)
+  }
+
+  private def startStateStoreCleaner(): Unit = {
+    val cleanerEnabled = conf.get(KyuubiConf.SERVER_STATE_STORE_CLEANER_ENABLED)
+    val stateMaxAge = conf.get(SERVER_STATE_STORE_MAX_AGE)
+
+    if (cleanerEnabled) {
+      val interval = conf.get(KyuubiConf.SERVER_STATE_STORE_CLEANER_INTERVAL)
+      val cleanerTask: Runnable = () => {
+        try {
+          _stateStore.cleanupMetadataByAge(stateMaxAge)
+        } catch {
+          case e: Throwable => error("Error cleaning up the metadata by age", e)
+        }
+      }
+
+      stateStoreCleaner.scheduleWithFixedDelay(
+        cleanerTask,
+        interval,
+        interval,
+        TimeUnit.MILLISECONDS)
+    }
+  }
+}
+
+object SessionStateStore {
+  @volatile private var _sessionStateStore: SessionStateStore = null
+
+  private[kyuubi] def set(sessionStateStore: SessionStateStore): Unit = {
+    _sessionStateStore = sessionStateStore
+  }
+
+  def get(): SessionStateStore = {
```

Comment:
where to use ?

Reply:
Removed it, it is used to init session state store in RestFe before.